components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -516,9 +516,16 @@ private void updateInboundAuthRequestConfiguration(int applicationId,
                     authKey = authRequest.getInboundAuthKey();
                     propertyArrayList = filterEmptyProperties(propertiesArray);
                 } else {
+                    String configType="";
+                    for (Property prop : propertiesArray) {
+                        if (StringUtils.equals(prop.getName(), ApplicationConstants.WELLKNOWN_APPLICATION_TYPE)) {
+                            configType = prop.getValue();
+                            break;
+                        }
+                    }
                     AbstractInboundAuthenticatorConfig inboundAuthenticatorConfig =
-                            ApplicationManagementServiceComponentHolder
-                                    .getInboundAuthenticatorConfig(authRequest.getInboundAuthType());
+                            ApplicationManagementServiceComponentHolder.getInboundAuthenticatorConfig(authRequest
+                                    .getInboundAuthType() + ":" + configType);
```

Comment:
Try to avoid the appending here. If there is a ":" in one of the string it will break.